scripts/airgapped/get-all-images.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/bash
+#
+# This script parses given bundle file for github repositories and branches. Then checks out each
+# charm's repository one by one using specified branch and collects images referred by that charm
+# using that repository's image collection script
+#
+BUNDLE_FILE=$1
+IMAGES=()
+# retrieve all repositories and branches for CKF
+REPOS_BRANCHES=($(yq -r '.applications[] | to_json' $BUNDLE_FILE | jq -r 'select(._github_repo_name) | "\(._github_repo_name):\(._github_repo_branch)"' | sort --unique))
+RESOURCE_DISPATCHER_BRANCH=track/1.0
+RESOURCE_DISPATCHER_REPO=https://github.com/canonical/resource-dispatcher
+
+for REPO_BRANCH in "${REPOS_BRANCHES[@]}"; do
+  IFS=: read -r REPO BRANCH <<< "$REPO_BRANCH"
+  git clone --branch $BRANCH https://github.com/canonical/$REPO
+  cd $REPO
+  IMAGES+=($(bash ./tools/get-images.sh))
+  cd - > /dev/null
+  rm -rf $REPO
+done
+
+# retrieve all repositories and branches for dependencies
+DEP_REPOS_BRANCHES=($(yq -r '.applications[] | to_json' $BUNDLE_FILE | jq -r 'select(._github_dependency_repo_name) | "\(._github_dependency_repo_name):\(._github_dependency_repo_branch)"' | sort --unique))
+
+for REPO_BRANCH in "${DEP_REPOS_BRANCHES[@]}"; do
+  IFS=: read -r REPO BRANCH <<< "$REPO_BRANCH"
+  git clone --branch $BRANCH https://github.com/canonical/$REPO
+  cd $REPO
+  # for dependencies only retrieve workload containers from metadata.yaml
+  IMAGES+=($(find -type f -name metadata.yaml -exec yq '.resources | to_entries | .[] | .value | ."upstream-source"' {} \;))
```

Comment:
In a different context I was using `yq eval '.resources | to_entries | map(select(.value.upstream-source != null) ...` as part of a pipeline.
Not sure at first glance if it's any better.

Reply:
Pushed a commit for this
https://github.com/canonical/bundle-kubeflow/pull/682/commits/88ad8428fe4a47fa17d627c257292f96710fdbf4